src/librustc_mir/hair/pattern/_match.rs

```diff
@@ -1418,7 +1418,8 @@ fn slice_pat_covered_by_const<'tcx>(
                 return Ok(false);
             }
             let n = n.assert_usize(tcx).unwrap();
-            alloc.get_bytes(&tcx, ptr, Size::from_bytes(n)).unwrap()
+            alloc.get_bytes(&tcx, ptr, Size::from_bytes(n),
+                            CheckInAllocMsg::OutOfBounds).unwrap()
```

Comment:
I think get_bytes doesn't need a `CheckInAllocMsg` argument, but can assume `MemoryAccess` inside the body

Reply:
fixed